src/main/java/org/kohsuke/github/GHWorkflowRun.java

```diff
@@ -261,6 +265,55 @@ public void rerun() throws IOException {
         root.createRequest().method("POST").withUrlPath(getApiRoute(), "rerun").fetchHttpStatusCode();
     }
 
+    /**
+     * Gets the artifacts associated with this workflow run.
+     *
+     * @return the paged iterable
+     */
+    public PagedIterable<GHArtifact> getArtifacts() {
+        try {
+            GitHubRequest request = root.createRequest().withUrlPath(getApiRoute(), "artifacts").build();
+            return new GHArtifactsIterable(owner, request);
+        } catch (MalformedURLException e) {
+            throw new GHException("Malformed URL", e);
+        }
+    }
+
+    /**
+     * Downloads the logs.
+     * <p>
+     * The logs are in the form of a zip archive. The full log file is at the root and called {@code 1_build.txt}. Split
+     * log files are also available in the {@code build} directory.
+     *
+     * @param <T>
+     *            the type of result
+     * @param streamFunction
+     *            The {@link InputStreamFunction} that will process the stream
+     * @throws IOException
+     *             The IO exception.
+     * @return the result of reading the stream.
+     */
+    public <T> T downloadLogs(InputStreamFunction<T> streamFunction) throws IOException {
+        requireNonNull(streamFunction, "Stream function must not be null");
+
+        String logsLocation = root.createRequest()
+                .method("GET")
+                .withUrlPath(getApiRoute(), "logs")
+                .fetchRedirectLocation();
```

Comment:
Interesting.  It doesn't automatically redirect? I think this one does.  
https://github.com/hub4j/github-api/blob/master/src/test/resources/org/kohsuke/github/GHRepositoryTest/wiremock/testZipball/mappings/repos_hub4j-test-org_temp-testzipball_zipball-3.json

 I'm surprised I feel sure we've had other endpoints that redirect. Could you test this with the okhttp3 connector?

Reply:
No, it doesn't. I tried without this magic as I thought it would work and I had an empty content in return. And I was doing the exact same thing than the code you pointed out.

But are you sure the redirect is working in the example you showed? Because there are no recorded request/response after that and I would expect the request to fail if not recorded.

Moreover the test would work even with an empty content.

Reply:
Apparently, the fact that it didn't follow redirects was a direct consequence of things not being implemented in Wiremock. The issue went away when I implemented the Wiremock server (well, indeed, the tests were broken because suddently the redirect was properly followed).